ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/Genesis/Tests/LongRangeAttack.hs

```diff
@@ -8,47 +9,85 @@
 import qualified Ouroboros.Network.AnchoredFragment as AF
+import           System.Random.Stateful (runSTGen_)
 import           Test.Consensus.Genesis.Setup
 import           Test.Consensus.Genesis.Setup.Classifiers
+import           Test.Consensus.PeerSimulator.Run (noTimeoutsSchedulerConfig)
+import           Test.Consensus.PeerSimulator.StateView
 import           Test.Consensus.PointSchedule
 import qualified Test.QuickCheck as QC
 import           Test.QuickCheck
 import           Test.QuickCheck.Extras (unsafeMapSuchThatJust)
+import           Test.QuickCheck.Random (QCGen)
 import           Test.Tasty
 import           Test.Tasty.QuickCheck
 import           Test.Util.Orphans.IOLike ()
 import           Test.Util.TestBlock (TestBlock, unTestHash)
 
 tests :: TestTree
-tests = testProperty "long range attack" prop_longRangeAttack
+tests =
+  testGroup "long range attack" [
+    testProperty "one adversary" (prop_longRangeAttack 1 [10])
+    -- TODO we don't have useful classification logic for multiple adversaries yet – if a selectable
+    -- adversary is slow, it might be discarded before it reaches critical length because the faster
+    -- ones have served k blocks off the honest chain if their fork anchor is further down the line.
+    -- ,
+    -- testProperty "three adversaries" (prop_longRangeAttack 1 [2, 5, 10])
+  ]
 
-genChainsAndSchedule :: Word -> ScheduleType -> QC.Gen (GenesisTest, PointSchedule)
-genChainsAndSchedule numAdversaries scheduleType =
+genChainsAndSchedule :: PointScheduleConfig -> Word -> ScheduleType -> QC.Gen (GenesisTest, PointSchedule)
+genChainsAndSchedule scheduleConfig numAdversaries scheduleType =
   unsafeMapSuchThatJust do
     gt <- genChains numAdversaries
-    pure $ ((gt,) <$> genSchedule scheduleType (gtBlockTree gt))
+    seed :: QCGen <- arbitrary
+    pure ((gt,) <$> runSTGen_ seed (\ g -> genSchedule g scheduleConfig scheduleType (gtBlockTree gt)))
 
-prop_longRangeAttack :: QC.Gen QC.Property
-prop_longRangeAttack = do
-  (genesisTest, schedule) <- genChainsAndSchedule 1 FastAdversary
+newLRA :: Bool
+newLRA = True
+
+prop_longRangeAttack :: Int -> [Int] -> QC.Gen QC.Property
+prop_longRangeAttack honestFreq advFreqs = do
+  (genesisTest, schedule) <- genChainsAndSchedule scheduleConfig (fromIntegral (length advFreqs)) sched
   let cls = classifiers genesisTest
 
-  pure $ withMaxSuccess 10 $ runSimOrThrow $
-    runTest genesisTest schedule $ \fragment ->
-        classify (genesisWindowAfterIntersection cls) "Full genesis window after intersection"
-        -- This is the expected behavior of Praos to be reversed with Genesis.
-        -- But we are testing Praos for the moment
-        $ existsSelectableAdversary cls ==> not $ isHonestTestFragH fragment
-        -- TODO
-        -- $ not (existsSelectableAdversary cls) ==> immutableTipBeforeFork fragment
+  -- TODO: not existsSelectableAdversary ==> immutableTipBeforeFork svSelectedChain
+
+  pure $ withMaxSuccess 10 $
```

Comment:
10 tests seem extremely low. I guess it's because the generators are, but I wonder how far we could push this. Also, we might want to make this configurable so that more tests are run in the CI and nightly setups.